app/controlplane/plugins/core/webhook/v1/webhook.go

```diff
@@ -0,0 +1,345 @@
+// File: app/controlplane/plugins/core/webhook/v1/webhook.go
+
+//
+// Copyright 2024 Shebash.io
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/app/controlplane/plugins/sdk/v1"
+	"github.com/go-kratos/kratos/v2/log"
+)
+
+// Integration implements a generic webhook integration
+type Integration struct {
+	*sdk.FanOutIntegration
+	client *http.Client
+}
+
+const (
+	providerWebhook = "webhook"
+)
+
+// registrationRequest defines the configuration required during registration
+type registrationRequest struct {
+	URL string `json:"url" jsonschema:"minLength=1,description=Webhook URL to send payloads to"`
+}
+
+// attachmentRequest defines the configuration required during attachment
+type attachmentRequest struct {
+	Materials string `json:"materials,omitempty" jsonschema:"description=Comma-separated list of materials to send (e.g., sbom, attestation)"`
```

Comment:
Sending more than just an attestation is a good idea but I think this option could be misleading since we do not support any kind of material type, just sbom. 

Because of that, I'd rather that we add a specific boolean option called `send-sbom` or similar.

In fact, to be consistent I think we could have two options

- send-attestation (default=true)
- send-sbom (default=false)

that way if you do not provide any settings you'll still have a sensible default.

What do you think?